src/ceph.in

```diff
@@ -749,10 +749,17 @@ def maybe_daemon_command(parsed_args, childargs):
             if childargs[1].find('/') >= 0:
                 sockpath = childargs[1]
             else:
+                _name = childargs[1]
+
                 # try resolve daemon name
+                if "client.admin" in childargs[1]:
```

Comment:
this workaround should be implemented in `ceph_conf()` as it is a limitation of `ceph-conf` which pretends to be another service. but in the case where "$pid" is included in the asock path, it fails.

also, it'd be better to use a list for known ceph daemons, instead of using a check just for "client.admin". like

```py
if any(service_type in name for service_type in ['mon', 'osd', 'mds', mgr']):
  # some long-running application includes $pid in its "admin socket",
  # while ceph-conf always use its own pid to substitute this meta variable
  # use the one provided in the specified name, if the given name does not look
  # like a daemon.
  # parse a name like client.admin.133423
  name, pid = name.rsplit('.', 1)[-1]
```

might want to catch `ValueError` in case user passes an invalid "name". and please don't use "_name" for name. use "name" instead.

Reply:
There maybe has other problems if we set the config likes:

```
 [client]
    socket admin = /run/ceph/ceph-$name.$pid.XXX.log                                                                                                                                  
```
Or
```
 [client]
    socket admin = /run/ceph/ceph.$pid.$name.log                                                                                                                                  
```

We cannot just trip the `$pid` off and we don't know which one will be the `$name`. Then this approach won't work.

The following is what I saw in one playbook deployed env:

```
# ls /run/ceph/
ceph-client.rgw.lxbceph1.rgw0.1190.93940079907960.asok  ceph-mds.lxbceph1.asok  ceph-mgr.lxbceph1.asok  ceph-mon.lxbceph1.asok  ceph-osd.0.asok
```

Maybe we should just add one `--pid` option for `ceph-conf`, and then we can do it simply:

`$ ceph-conf --name client.admin --pid 133423 --show-config-value admin_socket`

Will it make sense ?

Reply:
i don't think it makes difference to pass `--pid` to `ceph-conf` instead of using a env var. it just more confusing to pass "pid" all the way down to the conf subsystem.

the crux of the problem is how "pid" is encoded in the argument of `name` option passed to `ceph` cli. in the example in [the tracker ticket ](https://tracker.ceph.com/issues/47977), 

> ./bin/ceph daemon client.admin.133423 config show

where you are using the notation of `$name.$pid` instead of `$type.$id`, and `$name` is "client.admin".  if you really want to use "socket admin" as the argument of `name` option instead of the `$name.$pid` notation. i'm afraid it's 

1. even more confusing.
1. difficult to implement.

Reply:
Yeah, agree.
How about in `ceph-conf` let's parse the `ceph.conf` file with the raw data to get to know what's the format it is and whether the `name` option passed to `ceph` cli has a daemon pid in it and where it is ? 

If there has then we will stipe the pid from the `name` option and set a `PID` env for it ?

Reply:
> How about in `ceph-conf` let's parse the `ceph.conf` file with the raw data to get to know what's the format it is

i am inclined to avoid doing this. as the `socket admin` is not supposed to be a user-facing handle of daemon or ceph utility.

can we just stick to `$type.$id` for daemons and `$name.$pid` for non daemons when passing the "name" to `ceph` as the argument of `daemon`?

Reply:
That means we will disable the `socket admin` options for both the daemons  like `mds/mon/osd,etc` and the case like the `client` ?